src/cpp/transport/tcp/RTCPMessageManager.cpp

```diff
@@ -91,7 +91,17 @@ size_t RTCPMessageManager::sendMessage(
 }
 
 bool RTCPMessageManager::sendData(
-        TCPChannelResource *p_channel_resource,
+        std::shared_ptr<TCPChannelResource>& channel,
```

Comment:
Using a reference instead of a copy can cause problems as the channel being freed while we are still using it.